src/setuptools_scm/_file_finders/git.py

```diff
@@ -22,8 +23,18 @@ def _git_toplevel(path: str) -> str | None:
         cwd = os.path.abspath(path or ".")
         res = _run(["git", "rev-parse", "HEAD"], cwd=cwd)
         if res.returncode:
+            # This catches you being in a git directory, but the
+            # permissions being incorrect.  With modern contanizered
+            # CI environments you can easily end up in a cloned repo
+            # with incorrect permissions and we don't want to silently
+            # ignore files.
+            if "--add safe.directory" in res.stderr:
+                log.error(res.stderr)
+                raise SystemExit(
+                    "git introspection failed: {}".format(res.stderr.split("\n")[0])
+                )
             # BAIL if there is no commit
-            log.error("listing git files failed - pretending there aren't any")
+            warnings.warn("listing git files failed - pretending there aren't any")
```

Comment:
this should stay a log error

Reply:
indeed ,this can be a log call as well - please note that theres significant change in the develop branch (splitting most of the implementations into the vcs-versioning package there=

Reply:
TBH I think these are probably too quiet; I'll leave this for now but perhaps surfacing the warnings better is food for thought on a new branch